yjit/src/core.rs

```diff
@@ -400,9 +400,110 @@ pub enum BranchShape {
     Default, // Neither target is next
 }
 
-// Branch code generation function signature
-type BranchGenFn =
-    fn(cb: &mut Assembler, target0: CodePtr, target1: Option<CodePtr>, shape: BranchShape) -> ();
+#[derive(Clone, Copy, Debug, Eq, PartialEq)]
+pub enum BranchGenFn {
+    Branchif(BranchShape),
+    Branchnil(BranchShape),
+    Branchunless(BranchShape),
+    JumpToTarget0(BranchShape),
+    JNZToTarget0,
+    JZToTarget0,
+    JBEToTarget0,
+    JITReturn,
+}
+
+impl BranchGenFn {
+    pub fn call(self, asm: &mut Assembler, target0: CodePtr, target1: Option<CodePtr>) {
+        match self {
+            BranchGenFn::Branchif(shape) => {
```

Comment:
```suggestion
            BranchGenFn::BranchIf(shape) => {
```

I feel like these names should be capitalized to match the naming convention but otherwise looks good 👍

Reply:
:+1: https://github.com/ruby/ruby/pull/7401/commits/980484c5bf476eb87f8c95ef20fe5c1f190f43f5